kernel/aster-nix/src/fs/utils/dentrymnt.rs

```diff
@@ -0,0 +1,191 @@
+            "." => self.this(),
+            ".." => self.effective_parent().unwrap_or(self.this()),
+            name => {
+                let children_dentry = self.dentry.lookup_fast(name);
+                match children_dentry {
+                    Some(dentry) => Self::new(self.mount_node().clone(), dentry.clone()),
+                    None => {
+                        let slow_dentry = self.dentry.lookup_slow(name)?;
+                        Self::new(self.mount_node().clone(), slow_dentry.clone())
+                    }
+                }
+            }
+        };
+        let dentrymnt = dentrymnt.overlaid_dentrymnt();
+        Ok(dentrymnt)
+    }
+
+    // Get the absolute path.
+    //
+    // It will resolve the mountpoint automatically.
+    pub fn abs_path(&self) -> String {
+        let mut path = self.effective_name();
+        let mut dir_dentrymnt = self.this();
+
+        loop {
+            match dir_dentrymnt.effective_parent() {
+                None => break,
+                Some(parent_dir_dentrymnt) => {
+                    path = {
+                        let parent_name = parent_dir_dentrymnt.effective_name();
+                        if parent_name != "/" {
+                            parent_name + "/" + &path
+                        } else {
+                            parent_name + &path
+                        }
+                    };
+                    dir_dentrymnt = parent_dir_dentrymnt;
+                }
+            }
+        }
+        debug_assert!(path.starts_with('/'));
+        path
+    }
+
+    /// Get the effective name of dentrymnt.
+    ///
+    /// If it is the root of mount, it will go up to the mountpoint to get the name
+    /// of the mountpoint recursively.
+    fn effective_name(&self) -> String {
+        if !self.dentry.is_root_of_mount() {
+            return self.dentry.name();
+        }
+
+        if self.mount_node.parent().is_some() & self.mount_node.mountpoint_dentry().is_some() {
+            let parent_dentrymnt = Self::new(
+                self.mount_node.parent().unwrap().upgrade().unwrap().clone(),
+                self.mount_node.mountpoint_dentry().unwrap().clone(),
+            );
+            parent_dentrymnt.effective_name()
+        } else {
+            self.dentry.name()
+        }
+    }
+
+    /// Get the effective parent of dentrymnt.
+    ///
+    /// If it is the root of mount, it will go up to the mountpoint to get the parent
+    /// of the mountpoint recursively.
+    fn effective_parent(&self) -> Option<Arc<Self>> {
+        if !self.dentry.is_root_of_mount() {
+            return Some(Self::new(
+                self.mount_node.clone(),
+                self.dentry.parent().unwrap().clone(),
+            ));
+        }
+        if self.mount_node.parent().is_some() & self.mount_node.mountpoint_dentry().is_some() {
+            let parent_dentrymnt = Self::new(
+                self.mount_node.parent().unwrap().upgrade().unwrap().clone(),
```

Comment:
Rewrite the big if-else block to the following form for improved efficiency and clarity.

```rust
let Some(parent) = self.mount_node.parent() else {
    return None;
}
let Some(mountpoint) = self.mount_node.mountpoint_dentry() else {
    return None;
}

let parent_dentrymnt = Self::new(parent.upgrade().unwrap(), mountpoint.clone())
parent_dentrymnt.effective_parent()

Reply:
Thank you very much for your suggestion, I have followed your guidance to rewrite the existing if-else logic. It not only reduces the indentation level of the code, but also improves the readability of the code, making the logical flow more intuitive.
And I will apply this technique to my other code to further improve the code's robustness and maintainability. Thank you for your valuable time and professional advice.